src/vsr/replica.zig

```diff
@@ -4390,58 +4429,34 @@ pub fn ReplicaType(
+                return;
+            }
 
-                const slot = self.journal.slot_for_op(op);
+            // Repair prepares in chronological order. Older prepares will be overwritten by the
+            // cluster earlier, so we prioritize their repair. This also encourages concurrent
+            // commit/repair.
+            var op = self.op_repair_min();
+            while (op <= self.op) : (op += 1) {
+                const slot = self.journal.slot_with_op(op).?;
                 if (self.journal.dirty.bit(slot)) {
-                    if (self.journal.slot_with_op(op) == null) {
-                        // If this op was between `commit_min` and `replica.op`, we would have
-                        // requested and repaired these headers.
-                        // If this op was between `op_checkpoint` and `commit_min`, we would have
-                        // a header (since we couldn't have committed otherwise).
-                        //
-                        // Therefore, this op must be either:
-                        // - less-than-or-equal-to `op_checkpoint` — we committed before
-                        //   checkpointing, but the entry in our WAL was found corrupt after
-                        //   recovering from a crash.
-                        // - or (indistinguishably) this might originally have been an op greater
-                        //   than replica.op, which was truncated, but is now corrupt.
-                        //
-                        // We don't try to repair this op because the slot belongs (or will soon
-                        // belong) to a newer op, from the new WAL wrap. Additionally, we may not
-                        // still have access to its surrounding commits to verify the hash chain.
-                        assert(op <= self.commit_min);
-                        assert(op <= self.op_checkpoint());
-                        assert(self.journal.faulty.bit(slot));
-
-                        log.debug("{}: repair_prepares: remove slot={} " ++
-                            "(faulty, precedes checkpoint)", .{
-                            self.replica,
-                            slot.index,
-                        });
-                        self.journal.remove_entry(slot);
-                        continue;
-                    }
-
-                    // If this is an uncommitted op, and we are the primary in `view_change` status,
+                    // This is an uncommitted op — if we are the primary in `view_change` status,
                     // then we will `request_prepare` from the cluster, set `nack_prepare_op`,
                     // and stop repairing any further prepares:
                     // This will also rebroadcast any `request_prepare` every `repair_timeout` tick.
                     if (self.repair_prepare(op)) {
                         if (self.nack_prepare_op) |nack_prepare_op| {
```

Comment:
This I think is another benefit of chronological repair! We `nack` the earliest prepare, so we only have one nacking round (with reverse-chronologicial, I think we might end up nacking a bunch of preps one-by-one?)